tests/test_validators.py

```diff
@@ -51,3 +62,56 @@ def test_valid(self):
     def test_invalid(self):
         with self.assertRaisesMessage(ValidationError, "Item b's value None is not of type int."):
             DictValuesTypeValidator(int)({"a": 42, "b": None})
+
+
+class FileValidatorTestCase(SimpleTestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        fd, cls.temporary_file_name = mkstemp()
+
+    @classmethod
+    def tearDownClass(cls):
+        os.remove(cls.temporary_file_name)
+
+    def test_valid_read(self):
+        FileValidator()(self.temporary_file_name)
+
+    def test_valid_write(self):
+        FileValidator(mode="w")(self.temporary_file_name)
+
+    @patch("appsettings.validators.open", side_effect=IOError(errno.ENOENT, "Not a file"))
+    def test_file_does_not_exist(self, open_mock):
+        with self.assertRaisesRegex(ValidationError, r"File .* does not exist\."):
+            FileValidator()(self.temporary_file_name)
+
+    @patch("appsettings.validators.open", side_effect=IOError(errno.ENOENT, "Not a file"))
+    def test_file_does_not_exist_without_validation(self, open_mock):
+        FileValidator(exists=False)(self.temporary_file_name)
+
+    @patch("appsettings.validators.open", side_effect=IOError(errno.EACCES, "No required permissions"))
```

Comment:
We could actually modify the permissions of a test file.

Reply:
I am not sure how would you do it. I haven't found a way to mock permissions. Changing the file itself may result in impossibility to remove it, or not?

Reply:
I don't think so. To delete a file, you need a write permission on its directory.

Reply:
You are right, done. But I am not sure if is it the right way. Anything can go wrong and some ghost files and directories may remain in the file system (experienced during development).

Reply:
That seems rather unlikely.